### PR TITLE
libinput: use epoll to wait for events on the fd

### PR DESCRIPTION
**Pull request by lrusak**
This allows us to wait for an incoming event on the libinput fd instead of just sleeping.

### CODE REVIEW

**Comment by lrusak** [comment]
maybe @whot can drop a line if this is the best way to do it :smile_cat:

File: xbmc/platform/linux/input/LibInputHandler.cpp [hunk 1]
```diff
@@ -107,9 +108,30 @@ void CLibInputHandler::Start()
...
+
   while (!m_bStop)
   {
-    auto ret = libinput_dispatch(m_li);
+    epoll_wait(epollFd, &event, 1, -1);
+
+    ret = libinput_dispatch(m_li);
     if (ret < 0)
     {
       CLog::Log(LOGERROR, "CLibInputHandler::%s - libinput_dispatch failed: %s", __FUNCTION__, strerror(-errno));
```
Comment (whot) [inline xbmc/platform/linux/input/LibInputHandler.cpp]: you're leaking the `epollFd` here on error, this should be the usual `goto out:` or whatever other approach you prefer :)

**Review by whot — APPROVED** [review]
minor change for the leaking fd, otherwise looks good! I'm surprised you survived that long with a `sleep(10)` without any complaints :)

File: xbmc/platform/linux/input/LibInputHandler.cpp (hunk 1 above)
Comment (lrusak) [reply xbmc/platform/linux/input/LibInputHandler.cpp]: Thanks for catching that!

**Comment by lrusak** [comment]
I'm actually not sure why I didn't poll before... I was just reading over the documentation and read that polling the fd was what should be happening. :+1: Thanks for the review!

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

**Pull request merged**